tests/test_metrics.py

```diff
@@ -478,3 +694,129 @@ def _test_confidence_interval(self, metric, expected_ci_low, expected_ci_high):
+        # hence this test may fail on a PC
+        self._test_grouped_instance_confidence_interval(
+            metric=GroupNormCohensHStringContainment(),
+            expected_ci_low=-0.743586957620825,
+            expected_ci_high=-0.3908330554711398,
+        )
+
+        # F1-based scores
+        self._test_grouped_instance_confidence_interval(
+            metric=GroupMeanTokenOverlap(),
+            references=GROUPED_INSTANCE_REFERENCES_SHORT,
+            predictions=GROUPED_INSTANCE_PREDICTIONS_SHORT,
+            expected_global_result={
+                "group_mean_f1": 0.5,
+                "score": 0.5,
+                "score_name": "group_mean_f1",
+                "group_mean_f1_ci_low": 0.32222222222222224,
+                "group_mean_f1_ci_high": 0.7900160821100434,
+                "score_ci_low": 0.32222222222222224,
+                "score_ci_high": 0.7900160821100434,
+                "group_mean_precision": 0.5,
+                "group_mean_precision_ci_low": 0.32222222222222224,
+                "group_mean_precision_ci_high": 0.7900160821100434,
+                "group_mean_recall": 0.5,
+                "group_mean_recall_ci_low": 0.32222222222222224,
+                "group_mean_recall_ci_high": 0.7900160821100434,
+            },
+        )
+
+    def _test_grouped_instance_confidence_interval(
+        self,
+        metric,
+        expected_ci_low=0.0,
+        expected_ci_high=1.0,
+        references=GROUPED_INSTANCE_REFERENCES,
+        predictions=GROUPED_INSTANCE_PREDICTIONS,
+        expected_global_result=None,
+    ):
+        """Test the calculation of confidence intervals for a given metric with group_mean reduction."""
+        import numpy as np
+
+        outputs = apply_metric(
+            metric=metric,
+            predictions=predictions,
+            references=references,
+            additional_inputs=GROUPED_INSTANCE_ADDL_INPUTS,
+        )
+
+        group_score_name = "_".join(
+            [
+                "group",
+                metric.reduction_map["group_mean"]["agg_func"][0],
+                metric.main_score,
+            ]
+        )
+
+        if expected_global_result is None:
+            expected_global_result = {
+                f"{group_score_name}_ci_low": expected_ci_low,
+                f"{group_score_name}_ci_high": expected_ci_high,
+                "score_ci_low": expected_ci_low,
+                "score_ci_high": expected_ci_high,
+            }
+
+        global_result = outputs[0]["score"]["global"].copy()
+        logger.info(global_result)
+        for score_name, score_value in global_result.items():
+            if score_name in expected_global_result:
+                # Test that the output value is the same as the expected value
+                # allow for cases where value is NaN
+                if not isinstance(score_value, str):
+                    if np.isnan(expected_global_result[score_name]):
```

Comment:
if it isn't expected to have NaN in the tests perhaps this part is not needed? it seems by the coverage tool that these lines are never executed ..

Reply:
yes because it doesn't happen in the tests, but it can happed if the distribution is degenerate (a single unique value), which can happen even if there are no NaNs in the original scores.  I mentioned this in the resampling method.  I guess I don't need it in the test if it never happens, but it is something that can happen (since the equality testing doesn't work if there are NaNs).

Reply:
I see.. since its test code i suggest either removing it or, if we want to be very rigorous, add test values that cause NaN values to appear, and then this code will be used..